BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -78,7 +78,16 @@ public class AppProcessor {
             coordinator?.showAlert(.defaultAlert(title: Localizations.anErrorHasOccurred))
             return
         }
-        coordinator?.navigate(to: .tab(.vault(.vaultItemSelection(otpAuthModel))))
+
+        let vaultItemSelectionRoute = AppRoute.tab(.vault(.vaultItemSelection(otpAuthModel)))
+        guard let userId = try? await services.stateService.getActiveAccountId(),
+              !services.vaultTimeoutService.isLocked(userId: userId),
+              await !((try? services.vaultTimeoutService.hasPassedSessionTimeout(userId: userId)) ?? true)
+        else {
+            await coordinator?.handleEvent(.setAuthCompletionRoute(vaultItemSelectionRoute))
+            return
+        }
```

Comment:
⛏️ Could we change this a bit for readability, less parenthesis and no need for `!`:
```suggestion
        let vaultItemSelectionRoute = AppRoute.tab(.vault(.vaultItemSelection(otpAuthModel)))
        guard let userId = try? await services.stateService.getActiveAccountId(),
              !services.vaultTimeoutService.isLocked(userId: userId),
              await (try? services.vaultTimeoutService.hasPassedSessionTimeout(userId: userId)) == false
        else {
            await coordinator?.handleEvent(.setAuthCompletionRoute(vaultItemSelectionRoute))
            return
        }
```

Reply:
Yep!